src/akkudoktoreos/prediction/elecpriceakkudoktor.py

```diff
@@ -184,80 +151,116 @@ def _update_data(self, force_update: Optional[bool] = False) -> None:
         """
         # Get Akkudoktor electricity price data
         akkudoktor_data = self._request_forecast(force_update=force_update)  # type: ignore
-
+        assert self.start_datetime  # mypy fix
         # Assumption that all lists are the same length and are ordered chronologically
         # in ascending order and have the same timestamps.
-        values_len = len(akkudoktor_data.values)
-        if values_len < 1:
-            # Expect one value set per prediction hour
-            raise ValueError(
-                f"The forecast must have at least one dataset, "
-                f"but only {values_len} data sets are given in forecast data."
-            )
-
-        # Get cached 8day values
-        elecprice_cache_file = CacheFileStore().get(key="ElecPriceAkkudoktor8dayCache")
-        if elecprice_cache_file is None:
-            # Cache does not exist - create it
-            elecprice_cache_file = CacheFileStore().create(
-                key="ElecPriceAkkudoktor8dayCache",
-                until_datetime=to_datetime("infinity"),
-                suffix=".npy",
-            )
-            np.save(elecprice_cache_file, self.elecprice_8days)
-        elecprice_cache_file.seek(0)
-        self.elecprice_8days = np.load(elecprice_cache_file)
-
-        # Get elecprice_charges_kwh_kwh
-        charges_kwh = (
-            self.config.elecprice_charges_kwh if self.config.elecprice_charges_kwh else 0.0
+
+        # Get elecprice_charges_kwh in wh
+        charges_wh = (self.config.elecprice_charges_kwh or 0) / 1000
+
+        highest_orig_datetime = None  # newest datetime from the api after that we want to update.
+
+        for value in akkudoktor_data.values:
+            orig_datetime = to_datetime(value.start, in_timezone=self.config.timezone)
+            if highest_orig_datetime is None or orig_datetime > highest_orig_datetime:
+                highest_orig_datetime = orig_datetime
+
+            price_wh = value.marketpriceEurocentPerKWh / (100 * 1000) + charges_wh
+
+            existing_record = next((r for r in self.records if r.date_time == orig_datetime), None)
+            if existing_record:
+                # Update existing record
+                existing_record.elecprice_marketprice_wh = price_wh
+            else:
+                self.insert(
+                    0,
+                    ElecPriceDataRecord(date_time=orig_datetime, elecprice_marketprice_wh=price_wh),
+                )
+
+        # Generate history array for prediction
+        history = np.array(
+            [
+                record.elecprice_marketprice_wh
+                for record in sorted(self.records, key=lambda r: r.date_time)
+                if record.elecprice_marketprice_wh is not None
+                and record.date_time
+                < highest_orig_datetime  # make sure we only real data for the prediction, so cant be newer then data from the api.
+            ]
+        )
```

Comment:
Use `self.key_to_array(key="elecprice_marketprice_wh", end_datetime=highest_orig_datetime, fill_method="linear")` to get the array. This assures you get a value for every hour before highest_orig_datetime. None values will be interpolated.